bennu-renderers/src/main/java/org/fenixedu/bennu/core/presentationTier/renderers/autoCompleteProvider/UserAutoComplete.java

```diff
@@ -61,26 +61,28 @@
 
     @Override
     public Collection<User> getSearchResults(Map<String, String> argsMap, String value, int maxCount) {
-        Set<User> users = new HashSet<User>();
-        String[] values = StringNormalizer.normalize(value).split(" ");
-        for (User user : Bennu.getInstance().getUserSet()) {
-            if (user.getProfile() != null) {
-                final String normalizedUser = StringNormalizer.normalize(user.getDisplayName());
-                if (hasMatch(values, normalizedUser)) {
-                    users.add(user);
-                }
-            }
-        }
-        return users;
+
+        final Stream<User> users = Bennu.getInstance().getUserSet().stream();
+        String trimmedValue = value.trim();
+        final String[] input = StringNormalizer.normalize(trimmedValue).split(" ");
+        return users.filter(u -> match(input, u)).collect(Collectors.toSet());
+
     }
 
-    private boolean hasMatch(String[] input, String personNameParts) {
+    private boolean match(String[] values, User u) {
+
+        return (values.length == 1 && u.getUsername().equalsIgnoreCase(values[0]))
```

Comment:
Instead of only accepting an exact match, why don't we also accept sub-strings of the username, for example:
`hasMatch(values, StringNormalizer.normalize(u.getUsername())`
instead of:
` u.getUsername().equalsIgnoreCase(values[0])`